src/nimble.cc

```diff
@@ -45,37 +45,53 @@
 
 #include "nimble_block_material_interface_factory_base.h"
 #include "nimble_contact_interface.h"
-#include "nimble_material_factory.h"
 #include "nimble_main.h"
+#include "nimble_material_factory.h"
 #include "nimble_parser.h"
 
 #ifdef NIMBLE_HAVE_KOKKOS
 #include "nimble_kokkos_block_material_interface_factory.h"
 #include "nimble_kokkos_material_factory.h"
 #endif
 
+namespace details {
+
+int
+parseCommandLine(int argc, char** argv, nimble::EnvironmentFlags& myFlags);
+
+}
+
 int
 main(int argc, char* argv[])
 {
-  nimble::Parser myParser;
   //
-  // note: If the command line does not specify any parameter '--use-kokkos',
-  // '--use-tpetra', or '--use-vt', we could choose to specify one of them with
-  // either myParser.SetToUseKokkos() or myParser.SetToUseTpetra() or
-  // myParser.SetToUseVT()
+  // Quick parsing of command line to check whether Kokkos, Tpetra, or VT is used.
   //
-  nimble::NimbleInitializeAndGetInput(argc, argv, myParser);
+  nimble::EnvironmentFlags myFlags;
+  details::parseCommandLine(argc, argv, myFlags);
```

Comment:
Can you make this accessible from the nimble_main.h header? Otherwise, all looks good!